### PR TITLE
Get icenet and testchipip unit tests working

### PR DESCRIPTION
**Pull request by zhemao**
IceNet and TestChipIP previously had their own generators and testharnesses in order to run unittests. But now that generators can use configurations in other packages, that isn't necessary any more. Ideally, we'd be able to use the freechips.rocketchip.unittest TestHarness directly, but unfortunately it doesn't output the *.plusArgs file, which is necessary for verilator. 

**Related issue**: 

https://github.com/firesim/icenet/pull/24

https://github.com/ucb-bar/testchipip/pull/78

**Type of change**: new feature

**Impact**: other

**Release Notes**

Support for running IceNet and TestChipIP unit tests

### CODE REVIEW

**Comment by davidbiancolin** [comment]
I'd consider adding [a subset] of these unit tests to the CI.

File: generators/chipyard/src/main/scala/unittest/TestHarness.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package chipyard.unittest
+
+import chisel3._
+import freechips.rocketchip.config.Parameters
+
+class TestHarness(implicit val p: Parameters) extends Module {
+  val io = IO(new Bundle { val success = Output(Bool()) })
+  io.success := Module(new UnitTestSuite).io.finished
```
Comment (jerryz123) [inline generators/chipyard/src/main/scala/unittest/TestHarness.scala:8]: Can't this be done by setting `BuildSystem` to `Module(new UnitTestSuite)`, and using a `IOBinder` to connect `io.success` to `dut.io.finished`?

Comment (zhemao) [reply generators/chipyard/src/main/scala/unittest/TestHarness.scala:8]: But then I'd need to change the configuration inside testchipip and have it depend on chipyard or create one inside chipyard.

File: sims/vcs/Makefile [hunk 2]
```diff
@@ -56,7 +56,7 @@ VCS_CC_OPTS = \
 VCS_NONCC_OPTS = \
 	+lint=all,noVCDE,noONGS,noUI \
 	-error=PCWM-L \
-	-timescale=1ns/10ps \
+	-timescale=1ns/1ps \
```
Comment (abejgonzalez) [inline sims/vcs/Makefile:59]: Why is this change needed?

Comment (zhemao) [reply sims/vcs/Makefile:59]: It's for the ClockUtilTests. They need 1ps precision timing for the clock generators.
